pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -711,7 +711,9 @@ private void setupInput(ContextImpl contextImpl) throws Exception {
                         }.getType()), contextImpl);
             }
         } catch (Exception e) {
-            log.error("Source open produced uncaught exception: ", e);
+            log.error("Source open produced uncaught exception with ContextImpl of {}. "
+                            + "sourceSpec is {}. Exception is {}.",
+                    contextImpl.toString(), sourceSpec.toString(), e);
```

Comment:
calling `.toString()` isn't necessary since the logger will do that when you pass an object. There's another FAQ entry about that: http://slf4j.org/faq.html#string_contents